File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int left=0;
        int right=heights.length-1;

        int maxArea=0;

        while(left<right)
        {
            int width=right-left;

            int h=Math.min(heights[left],heights[right]);

            int area=width*h;

            maxArea=Math.max(maxArea,area);

            if(heights[left]<heights[right])
            {
                left++;
            }
            else
            {
                right--;
            }
        }
        return maxArea;

    }
}

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:

        stack=[]

        max_area=0

        for i in range(len(heights)):

            while stack and heights[i]<heights[stack[-1]]:
                height=heights[stack.pop()]

                if stack:
                    width=i-stack[-1]-1


                else:

                    width=i

                area=height*width

                max_area=max(max_area,area)

            stack.append(i)  

        while stack:
            height=heights[stack.pop()]

            if stack:
                width=len(heights)-stack[-1]-1

            else:
                width=len(heights)

            area=height*width

            max_area=max(max_area,area)

        return max_area

        
        